activerecord/lib/active_record/relation.rb

```diff
@@ -182,18 +182,19 @@ def find_or_create_by!(attributes, &block)
       find_by(attributes) || create!(attributes, &block)
     end
 
-    # Attempts to create a record with the given attributes in a table that has a unique constraint
+    # Attempts to create a record with the given attributes in a table that has a unique DB constraint
     # on one or several of its columns. If a row already exists with one or several of these
     # unique constraints, the exception such an insertion would normally raise is caught,
     # and the existing record with those attributes is found using #find_by!.
     #
     # This is similar to #find_or_create_by, but avoids the problem of stale reads between the SELECT
     # and the INSERT, as that method needs to first query the table, then attempt to insert a row
-    # if none is found.
+    # if none is found. The use of this method is suggested when #find_by! rarely returns a record,
```

Comment:
This new addition is unclear. Reading this I'm thinking when is a record rarely returned? I think what you mean is when you want to find or create use that otherwise use find. I don't think we use `find_or_create` when there's rarely a record, it's used for specific application behavior when you absolutely want a record to exist - usually in the case of a transaction rollback due to a uniqueness constraint.

Reply:
This is a poor rephrase of: 
>"I believe the intent of the method create_or_find_by is to get to get to the INSERT portion as fast as it can, and this is helpful for tables where find_by rarely returns a record (meaning you're going to INSERT much more than you're going to find the record). "

Honestly after re-reading I don't think this would add much value, if you don't mind I've just removed such addition.

Apologies for my lack of English skills 😞